docs/advanced/errors.md

```diff
@@ -19,10 +21,51 @@ export class MyErrorHandler implements ErrorHandler {
   imports: [AppComponent],
   providers: [
     {
-      provide: ErrorHandler, 
+      provide: ErrorHandler,
       useClass: MyErrorHandler
     }
   ]
 })
 export class AppModule { }
 ```
+
+## Handling errors within an @Action
+
+When you define an @Action you can handle error within the action and if you do so, the error will not propagate to Angular's global `ErrorHandler`, nor the `dispatch` Observable. This applies to both sync and async types of Actions.
+
+```TS
+  @Action(HandledError)
+  handledError({ getState, setState }) {
+    try {
+      // error is thrown
+    } catch (err) {
+      console.log("error catched inside @Action wont propagate to ErrorHandler or dispatch subscription")
+    }
+  }
+```
+
+## Handling errors after dispatching an action
+
+If an unhandled exception is thrown inside an action, the error will be propagated to the `ErrorHandler` and you can also catch it subscribing to the `dispatch` Observable. If you subscribe to the `dispatch` Observable the error will be caught twice, once in the ErrorHandler and on your `dispatch` handle.
+
+```TS
+  @Action(UnhandledError)
+  unhandledError({ getState, setState }) {
```

Comment:
`ctx` instead of `{ getState, setState })`

Reply:
Just for context, we prefer to not use destructuring too much in the docs because it assumes a level of knowledge from the reader. We aim for a little bit simpler syntax.

Reply:
yep

```ts
  @Action(UnhandledError)
  unhandledError(ctx: StateContext<StateModel>) {
```